packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -1168,6 +1169,26 @@ class ReactDOMServerRenderer {
           case REACT_EVENT_COMPONENT_TYPE:
           case REACT_EVENT_TARGET_TYPE: {
             if (enableEventAPI) {
+              if (
+                elementType.$$typeof === REACT_EVENT_TARGET_TYPE &&
+                elementType.type === REACT_EVENT_TARGET_TOUCH_HIT
+              ) {
+                const props = nextElement.props;
+                const bottom = props.bottom || 0;
+                const left = props.left || 0;
+                const right = props.right || 0;
+                const top = props.top || 0;
+
+                if (bottom === 0 && left === 0 && right === 0 && top === 0) {
+                  return '';
+                }
+                let topString = top ? `-${top}px;` : '';
+                let leftString = left ? `-${left}px;` : '';
+                let rightString = right ? `-${right}px;` : '';
+                let bottomString = bottom ? `-${bottom}px;` : '';
+
+                return `<div style="position:absolute;${topString}${leftString}${rightString}${bottomString}"></div>`;
```

Comment:
It's worth noting that this could result in an undesirable experience for mouse-based interactions if the client-side JS is available.

Reply:
Not sure how we tackle that one. Maybe @sebmarkbage had some ideas. I think you can maybe get away using some clever `pointer` media queries in a stylesheet.